pkg/server/app/commands/submit_transaction.go

```diff
@@ -29,30 +54,93 @@ type SubmitTransactionProvider interface {
 // into an overlay-engine-compatible format, and applying any other necessary
 // logic before invoking the engine.
 type SubmitTransactionHandler struct {
-	provider SubmitTransactionProvider
+	provider         SubmitTransactionProvider
+	requestBodyLimit int64
+}
+
+// CreateTaggedBEEFFromRequest extracts the topics from the header and reads the body
+// to create a TaggedBEEF object, with size limits applied.
+func (s *SubmitTransactionHandler) CreateTaggedBEEFFromRequest(r *http.Request) (*overlay.TaggedBEEF, error) {
+	header := r.Header.Get(XTopicsHeader)
+	if header == "" {
+		return nil, ErrMissingXTopicsHeader
+	}
+
+	var topics []string
+	if err := json.Unmarshal([]byte(header), &topics); err != nil {
+		return nil, ErrInvalidXTopicsHeaderFormat
+	}
+
+	reader := io.LimitReader(r.Body, s.requestBodyLimit)
+	buff := make([]byte, 64*1024) // typically it's a best balance between performance and memory usage (loop and size).
+	bytesRead := 0
+	for {
+		n, err := reader.Read(buff)
+		bytesRead += n
+
+		if bytesRead > int(s.requestBodyLimit) {
+			return nil, ErrRequestBodyTooLarge
+		}
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return nil, ErrRequestBodyRead
+		}
+	}
+
+	return &overlay.TaggedBEEF{Beef: buff, Topics: topics}, nil
 }
 
 // Handle orchestrates the processing flow of a transaction. It prepares and
 // sends a JSON response after invoking the engine and returns an HTTP response
 // with the appropriate status code based on the engine's response.
 func (s *SubmitTransactionHandler) Handle(w http.ResponseWriter, r *http.Request) {
-	// TODO: Add custom validation logic.
-	steak, err := s.provider.Submit(r.Context(), overlay.TaggedBEEF{}, engine.SubmitModeCurrent, func(steak *overlay.Steak) {})
+	defer func() {
+		if err := r.Body.Close(); err != nil {
+			fmt.Printf("Failed to close request body: %v\n", err)
+		}
+	}()
```

Comment:
```suggestion
	defer defer func() {
			_ = r.Body.Close()
		}()
```
this will pass linter

Reply:
ok thanks 👍